src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala

```diff
@@ -48,4 +54,125 @@ class EasyStageFileItemSpec extends FlatSpec with Matchers {
     val args = "src/test/resources/example.csv outdir".split(" ")
     getSettingsRows(new FileItemConf(args)).get.size shouldBe 5
   }
+
+  "main" should "report a configuration problem" in {
+    val args = "src/test/resources/example.csv target/testSDO".split(" ")
+    the[Exception] thrownBy EasyStageFileItem.main(args) should
+      have message "no protocol: {{ easy_stage_dataset_fcrepo_service_url }}"
+  }
+
+  "run" should "create expected file item SDOs" in {
+    EasyStageFileItem.run(new FileItemSettings(
+      sdoSetDir = Some(new File("target/testSDO")),
+      datastreamLocation = Some(new URL("http://x.nl/l/d")),
+      size = Some(1),
+      datasetId = Some("easy-dataset:1"),
+      pathInDataset = Some(new File("original/newSub/file.mpeg")),
+      format = Some("video/mpeg"),
+      subordinate = "object" -> s"info:fedora/easy-dataset:1",
+      easyFilesAndFolders = mockEasyFilesAndFolders(HashMap(
+        "easy-dataset:1 original/newSub/file.mpeg" -> Success("original", "easy-folder:1")
+      )),
+      fedora = mockFedora(HashMap(
+        "pid~easy-dataset:1" -> Seq("easy-dataset:1")
+      ))
+    ))
+
+    // comparing with sample output
+
+    // TODO sdoDir "newSub" should have been "original_newSub" to avoid potential conflicts !!!
+    val actualSdoSet = Path("target/testSDO/easy-dataset_1")
+    val expectedSdoSet = Path("src/test/resources/expectedFileItemSDOs")
+    getRelativeFiles(actualSdoSet) shouldBe getRelativeFiles(expectedSdoSet)
+    actualSdoSet.walk.toSeq.map(_.path).sortBy(s => s).zip(
+      expectedSdoSet.walk.toSeq.map(_.path).sortBy(s => s)
+    ).foreach {
+      case (actual, expected) if actual.endsWith("cfg.json") =>
+        readCfgJson(expected) shouldBe readCfgJson(actual)
+      case (actual, expected) if actual.endsWith("fo.xml") =>
+        readDatastreamFoxml(actual) shouldBe readDatastreamFoxml(expected)
+      case (actual, expected) => // metadata of a file or folder
+        readFlatXml(actual) shouldBe readFlatXml(expected)
+    }
+
+    // a less verbose check reworded
```

Comment:
@DANS-KNAW/easy 
The comment by Jan above applies to an earlier commit that compared the exact contents of the actual and expected files. I understand Jan wants to see checks as shown below (currently incomplete as it compares the content of just one file) but IMHO the expected output under the test resources makes it easier to picture what is produced and thus better documents the application. The current implementation is no longer sensitive for irrelevant changes such as white space and order of elements or attributes. The methods in [SdoFiles](https://github.com/DANS-KNAW/easy-stage-dataset/pull/62/files#diff-19a676551f79b0048cb9c643b8eded83R32) take care of that.